src/main/java/seedu/address/storage/JsonAdaptedAttendance.java

```diff
@@ -14,10 +32,83 @@
     @JsonCreator
     public JsonAdaptedAttendance(@JsonProperty("attendance date") String attendanceDate,
+                                 @JsonProperty("isPresent") String isPresent,
                                  @JsonProperty("Pick up time") String pickUpTime,
                                  @JsonProperty("Drop off time") String dropOffTime) {
         this.attendanceDate = attendanceDate;
+        this.isPresent = isPresent;
         this.pickUpTime = pickUpTime;
         this.dropOffTime = dropOffTime;
     }
+
+    /**
+     * Constructs a given {@code Attendance} into this class for Jackson use.
+     *
+     * @param attendanceEntry the attendance of the pet.
+     */
+    public JsonAdaptedAttendance(AttendanceEntry attendanceEntry) {
+
+        this.attendanceDate = attendanceEntry.getAttendanceDate().format(DateTimeFormatter.ISO_LOCAL_DATE);
+        this.isPresent = attendanceEntry.getIsPresent().get().toString();
+
+        Optional<LocalTime> optionalPickUpTime = attendanceEntry.getPickUpTime();
+        Optional<LocalTime> optionalDropOffTime = attendanceEntry.getDropOffTime();
+
+        this.pickUpTime = optionalPickUpTime.isEmpty()
+                ? ""
+                : optionalPickUpTime.map(t -> t.format(DateTimeFormatter.ISO_LOCAL_TIME)).get();
+        this.dropOffTime = optionalDropOffTime.isEmpty()
+                ? ""
+                : optionalDropOffTime.map(t -> t.format(DateTimeFormatter.ISO_LOCAL_TIME)).get();
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted attendance object into the model's {@code Attendance} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted attendance object.
+     */
+    public AttendanceEntry toModelType() throws IllegalValueException {
+
+        LocalDate modelAttendanceDate;
+        Boolean modelIsPresent;
+        LocalTime modelPickUpTime;
+        LocalTime modelDropOffTime;
+        PresentAttendanceEntry presentAttendanceEntry;
+
+        if (!isValidIsPresentString(isPresent)) {
+            throw new IllegalValueException(AttendanceEntry.MESSAGE_INVALID_ISPRESENT);
+        }
+
+        modelIsPresent = Boolean.parseBoolean(isPresent);
+
+        try {
+            modelAttendanceDate = AttendanceUtil.convertToModelDate(attendanceDate);
+        } catch (DateTimeParseException pe) {
+            throw new IllegalValueException(pe.getMessage());
+        }
+
+        if (!modelIsPresent) {
+            return new AbsentAttendanceEntry(modelAttendanceDate);
+        }
+
+        try {
+            modelPickUpTime = AttendanceUtil.convertToModelTime(pickUpTime);
+        } catch (DateTimeParseException pe) {
+            throw new IllegalValueException(pe.getMessage());
+        }
+
+        try {
+            modelDropOffTime = AttendanceUtil.convertToModelTime(dropOffTime);
+        } catch (DateTimeParseException pe) {
+            throw new IllegalValueException(pe.getMessage());
+        }
```

Comment:
Not sure if you can condense all the try catch blocks into one or is there a need to be sequential for them?

Reply:
Good idea!